src/core/Form/SearchInput/SearchInput.tsx

```diff
@@ -1,47 +1,265 @@
+import { idGenerator } from '../../../utils/uuid';
+import { TokensProp, InternalTokensProp } from '../../theme';
+import { withSuomifiDefaultProps } from '../../theme/utils';
+import { VisuallyHidden } from '../../../components/Visually-hidden/Visually-hidden';
+import { StatusText } from '../StatusText/StatusText';
+import { LabelText, LabelMode } from '../LabelText/LabelText';
+import { Icon } from '../../Icon/Icon';
+import { baseStyles } from './SearchInput.baseStyles';
+
+type StateValue = string | number | string[] | undefined;
+
+type SearchInputStatus = 'default' | 'error';
+
+export interface SearchInputProps
+  extends Omit<
+      HtmlInputProps,
+      'type' | 'disabled' | 'onChange' | 'children' | 'onClick'
+    >,
+    TokensProp {
+  /** SearchInput container div class name for custom styling. */
+  className?: string;
+  /** SearchInput container div props */
+  inputContainerProps?: Omit<HtmlDivProps, 'className'>;
+  /** Label text */
+  labelText: string;
+  /** Hide or show label. Label element is always present, but can be visually hidden.
+   * @default visible
+   */
+  labelMode?: LabelMode;
+  /** Placeholder text for input. Use only as visual aid, not for instructions. */
+  visualPlaceholder?: string;
+  /** Clear button label for screen readers */
+  clearButtonLabel: string;
+  /** Search button label for screen readers */
+  searchButtonLabel: string;
+  /** SearchButtonProps */
+  searchButtonProps?: Omit<HtmlButtonProps, 'onClick' | 'tabIndex'>;
+  /**
+   * 'default' | 'error'
+   * @default default
+   */
+  status?: SearchInputStatus;
+  /** Status text to be shown below the component and hint text. Use e.g. for validation error */
+  statusText?: string;
+  /** Input name */
+  name?: string;
+  /** Set components width to 100% */
+  fullWidth?: boolean;
+  /** Callback for input text change */
+  onChange?: (value: StateValue) => void;
+  /** Callback for onBlur event */
+  onBlur?: (event: FocusEvent<HTMLInputElement>) => void;
+  /** Callback for search button click */
+  onSearch?: (value: StateValue) => void;
+}
 
 const baseClassName = 'fi-search-input';
-const iconBaseClassName = `${baseClassName}_icon`;
+const searchInputClassNames = {
+  fullWidth: `${baseClassName}--full-width`,
+  error: `${baseClassName}--error`,
+  notEmpty: `${baseClassName}--not-empty`,
+  inputElement: `${baseClassName}_input`,
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  functionalityContainer: `${baseClassName}_functionality-container`,
+  button: `${baseClassName}_button`,
+  searchButton: `${baseClassName}_button-search`,
+  searchIcon: `${baseClassName}_button-search-icon`,
+  clearButton: `${baseClassName}_button-clear`,
+  clearIcon: `${baseClassName}_button-clear-icon`,
+};
 
-export interface SearchInputProps extends Omit<TextInputProps, 'status'> {
-  status?: 'default' | 'error';
+interface SearchInputState {
+  value: string | number | string[] | undefined;
```

Comment:
Could the `StateValue` type  used here? 
It seems to have the same possible values listed and name indicates that it is the same thing.

Reply:
Yeah, should also probably rename it since it's exported with the interface. Maybe SearchInputStateValue etc.

Reply:
Renamed to SearchInputValue and also used for the line above.